staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/defaulting/validation.go

```diff
@@ -96,6 +98,11 @@ func validate(ctx context.Context, pth *field.Path, s *structuralschema.Structur
 				if remainingCost < 0 {
 					return allErrs, nil, remainingCost
 				}
+
+				if len(celErrs) == 0 && utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CRDValidationRatcheting) {
```

Comment:
needs a comment explaining why we're calling this again

also, I think we need to capture the cost of both types of runs and take whichever leaves less remaining cost

```go
celErrs, rmCost := celValidator.Validate(ctx, pth.Child("default"), s, s.Default.Object, s.Default.Object, remainingCost)
allErrs = append(allErrs, celErrs...)

if len(celErrs) == 0 && utilfeature.DefaultFeatureGate.Enabled(apiextensionsfeatures.CRDValidationRatcheting) {
  celErrs, rmCostWithoutOldObject := celValidator.Validate(ctx, pth.Child("default"), s, s.Default.Object, nil, remainingCost)
  allErrs = append(allErrs, celErrs...)
  if rmCostWithoutOldObject < rmCost {
    rmCost = rmCostWithoutOldObject
  }
}

remainingCost = rmCost
if remainingCost < 0 {
  return allErrs, nil, remainingCost
}
```

Reply:
same comment applies below

Reply:
Applied feedback as suggested and added comments